flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -661,10 +682,117 @@ public void testConcurrentGlobalFailAndRestarts() throws Exception {
 		}
 	}
 
+	@Test
+	public void testRestartWithEagerSchedulingAndSlotSharing() throws Exception {
```

Comment:
Exactly, it extends test coverage to possible related conditions.
I realized that we had no test about the "happy path" of eager scheduling and slot sharing as well.